x-pack/plugins/fleet/server/services/security/uninstall_token_service/index.ts

```diff
@@ -208,13 +214,19 @@ export class UninstallTokenService implements UninstallTokenServiceInterface {
       return [];
     }
 
-    const filter: string = tokenObjectHits
-      .map(({ _id }) => {
-        return `${UNINSTALL_TOKENS_SAVED_OBJECT_TYPE}.id: "${_id}"`;
-      })
-      .join(' or ');
+    const filterEntries: string[] = tokenObjectHits.map(
+      ({ _id }) => `${UNINSTALL_TOKENS_SAVED_OBJECT_TYPE}.id: "${_id}"`
+    );
+
+    const uninstallTokenChunks: UninstallToken[][] = await asyncMap(
+      chunk(filterEntries, this.batchSize),
+      (entries) => {
+        const filter = entries.join(' or ');
+        return this.getDecryptedTokens({ filter });
+      }
+    );
 
-    return this.getDecryptedTokens({ filter });
+    return uninstallTokenChunks.flat();
```

Comment:
Is there any performance concern we should have here if the `uninstallTokenChunks` is too large for `flat` method?

Reply:
I don't think it can be too large, for that we would need to have 2^32 order of magnitude number of policies.

but in overall, I think we should take some time to improve the performance of token validation in Fleet Setup